base-ci-resource.yaml

```diff
@@ -104,6 +106,24 @@ testNodeApplicationUsingPostgres:
     POSTGRES_SERVER: postgres
     POSTGRES_PASSWORD: ''
 
+# If project is a node typescript app it will use this job configuration for build and testing
+testNodeTypescriptApplication:
+  extends: .node
+  image: briangweber/docker-node:carbon
+  stage: build
+  only:
+    variables:
+      - $NODEPOSTGRESAPP == null
+      - $NODETYPESCRIPTAPP == "true"
+  script:
+    - npm run build
```

Comment:
This is the only different line, right? And it is only one new build step because we don't currently have a ts app with a db server for tests?

Reply:
this line, plus `dist/` under artifacts. I suppose we could use base node job + inheritance to support typescript

Reply:
nested inheritance is nice in the latest gitlab via `extends` but I don't believe we have it yet